ipa-core/src/protocol/basics/mul/semi_honest.rs

```diff
@@ -73,24 +73,24 @@ where
 {
     let role = ctx.role();
 
-    // Compute the value (d_i) we want to send to the right helper (i+1).
-    let right_d =
-        a.left_arr().clone() * b.right_arr() + a.right_arr().clone() * b.left_arr() - prss_left;
-
-    ctx.send_channel::<<F as Vectorizable<N>>::Array>(role.peer(Direction::Right))
-        .send(record_id, &right_d)
+    // Compute the value z_i we want to send to the left helper, i.e. (i-1).
+    let z_left = a.left_arr().clone() * b.left_arr()
```

Comment:
why is it in this change? Can you provide description for this PR that explains the changes here?